core/node/groups.go

```diff
@@ -281,6 +283,7 @@ func Online(bcfg *BuildCfg, cfg *config.Config) fx.Option {
 func Offline(cfg *config.Config) fx.Option {
 	return fx.Options(
 		fx.Provide(offline.Exchange),
+		fx.Provide(DNSResolver),
```

Comment:
This preserves existing behavior. @Stebalien Is this the correct behavior, or would we want some `nilDNSResolver` here?

Reply:
Yes, we should use the `nilDNSResolver`. I'd consider this a bug fix.

Reply:
But we can also punt to a second PR.

Reply:
It's pretty easy to do, so I can do it here.

Reply:
done.

Reply:
undone, as it is a behaviour change that leads to a regression; we can easily consider it in a follow up pr.